### ISSUE TITLE
ClassSamplingFactors Bug, java.lang.Double cannot be cast to java.lang.Float

### ISSUE THREAD

**Issue by elunenule** [h2oai/sparkling-water#2418]
Hello, thank you in advance for the support you regularly provide!

I was testing Sparkling Water and i wanted to try out class balancing using a DRF model when i encountered the following issue
```
py4j.protocol.Py4JJavaError: An error occurred while calling o471.w.
: java.lang.ClassCastException: java.lang.Double cannot be cast to java.lang.Float
	at ai.h2o.sparkling.ml.params.NullableFloatArrayParam$$anonfun$w$1.apply(NullableFloatArrayParam.scala:34)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at ai.h2o.sparkling.ml.params.NullableFloatArrayParam.w(NullableFloatArrayParam.scala:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:280)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:748)
```

#### Sparkling Water/PySparkling  version
Python_version:             3.6.8 final
Sparkling Water Context:
 * Sparkling Water Version: 3.32.0.2-1-2.2
 * H2O name: yarn
 * cluster size: 5
py4j-0.10.4
#### Hadoop Version & Distribution
* HDP 2.6.5 Hortonworks
#### Yarn Logs
https://github.com/elunenule/logs/blob/master/launch_container.log
https://github.com/elunenule/logs/blob/master/stderr.log
https://github.com/elunenule/logs/blob/master/stdout.log
#### Spark & Sparkling Water configuration including the memory configuration
```
(spark.scheduler.minRegisteredResourcesRatio,1)
    (spark.ext.h2o.backend.cluster.mode,internal)
    (spark.yarn.app.container.log.dir,/var/log/hadoop/yarn/log/application_1607707701059_0024/container_e72_1607707701059_0024_01_000001)
    (spark.task.maxFailures,1)
    (spark.yarn.appMasterEnv.PEX_ALWAYS_CACHE,false)
    (spark.serializer,org.apache.spark.serializer.KryoSerializer)
    (spark.pyspark.python,./venv-0.1.dev85+g4c85d90.pex)
    (spark.yarn.am.memoryOverhead,300m)
    (spark.ext.h2o.repl.enabled,false)
    (spark.executorEnv.PEX_INHERIT_PATH,false)
    (spark.yarn.dist.files,file:/home/atigbad/projects/configs/etl_config.json,file:/home/atigbad/projects/./dist/venv-0.1.dev85+g4c85d90.pex)
    (spark.yarn.am.extraJavaOptions,-verbose:gc -XX:+PrintGCTimeStamps -XX:+PrintGCDetails -XX:+UseCompressedOops -XX:NewSize=3g)
    (spark.locality.wait,0)
    (spark.ext.h2o.client.language,python)
    (spark.app.name,Modelisation Dec-13-2020:12:29)
    (spark.ui.filters,org.apache.hadoop.yarn.server.webproxy.amfilter.AmIpFilter)
    (spark.driver.memory,14g)
    (spark.executor.memory,14g)
    (spark.ext.h2o.nthreads,-1)
    (spark.ext.h2o.cluster.size,5)
    (spark.yarn.appMasterEnv.PEX_VERBOSE,0)
    (spark.config.debug_level,info)
    (spark.hadoop.yarn.timeline-service.enabled,false)
    (spark.yarn.appMasterEnv.PEX_UNZIP,true)
    (spark.executor.id,driver)
    (spark.yarn.am.memory,27g)
    (spark.yarn.executor.memoryOverhead,300m)
    (spark.yarn.appMasterEnv.PEX_PROFILE,false)
    (spark.app.id,application_1607707701059_0024)
    (spark.rdd.compress,true)
    (spark.executorEnv.PEX_PROFILE,false)
    (spark.executorEnv.PYTHONPATH,{{PWD}}/pyspark.zip<CPS>{{PWD}}/py4j-0.10.4-src.zip)
    (spark.driver.extraJavaOptions,-verbose:gc -XX:+PrintGCTimeStamps -XX:+PrintGCDetails -XX:+UseCompressedOops -XX:NewSize=3g)
    (spark.yarn.max.executor.failures,1)
    (spark.driver.port,40683)
    (spark.ext.h2o.hdfs_conf,/data1/yarn/local/usercache/atigbad/appcache/application_1607707701059_0024/spark-7a67dc54-b408-423a-a071-ff359fbf7ec6/hdfs_conf1874663290513014225.xml)
    (spark.yarn.app.id,application_1607707701059_0024)
    (spark.executor.extraJavaOptions,-verbose:gc -XX:+PrintGCTimeStamps -XX:+PrintGCDetails -XX:+UseCompressedOops -XX:NewSize=3g)
    (spark.submit.deployMode,cluster)
    (spark.hadoop.yarn.client.failover-proxy-provider,org.apache.hadoop.yarn.client.ConfiguredRMFailoverProxyProvider)
    (spark.ext.h2o.cloud.name,sparkling-water-yarn_application_1607707701059_0024)
    (spark.pyspark.driver.python,./venv-0.1.dev85+g4c85d90.pex)
    (spark.executorEnv.PEX_VERBOSE,0)
    (spark.config.env,dev)
    (spark.jars.packages,)
    (spark.yarn.appMasterEnv.PEX_ROOT,./)
    (spark.executorEnv.PEX_ROOT,./)
    (spark.serializer.objectStreamReset,100)
    (spark.driver.cores,1)
    (spark.driver.host,172.30.2.44)
    (spark.yarn.appMasterEnv.PEX_INHERIT_PATH,false)
    (spark.executorEnv.PEX_ALWAYS_CACHE,false)
    (spark.executorEnv.PEX_UNZIP,true)
    (spark.org.apache.hadoop.yarn.server.webproxy.amfilter.AmIpFilter.param.PROXY_HOSTS,datalab-hdp-master-2.novalocal,datalab-hdp-master-1.novalocal)
    (spark.logConf,true)
    (spark.master,yarn)
    (spark.ui.port,0)
    (spark.executor.instances,5)
    (spark.yarn.driver.memoryOverhead,300m)
    (spark.org.apache.hadoop.yarn.server.webproxy.amfilter.AmIpFilter.param.PROXY_URI_BASES,http://datalab-hdp-master-2.novalocal:8088/proxy/application_1607707701059_0024,http://datalab-hdp-master-1.novalocal:8088/proxy/application_1607707701059_0024)
    (spark.yarn.isPython,true)
    (spark.executor.cores,1)
```

#### Please also provide us with the full and minimal reproducible code.
I cannot really provide a sample dataset due to company policy however the labelCol contains 5 classes.
```python
from pysparkling.ml import H2OGBM, H2OGBMClassifier, H2ODRF
        drf_estimator = H2ODRF(  # featuresCols=features.to_list(),
            featuresCols=features.to_list(),
            ignoredCols=['dataJeu'],
            labelCol="PROJETIndex",
            columnsToCategorical=string_columns,
            modelId="drf_test2",
            splitRatio=0.8,
            predictionCol="prediction",
            balanceClasses=True,
	    maxAfterBalanceSize=0.9,
	    classSamplingFactors=[0.02, 1, 1, 1, 1],
	    categoricalEncoding="OneHotExplicit")
```

**Comment by elunenule** [comment]
I have tried with `classSamplingFactors=[0.02, 1.0, 1.0, 1.0, 1.0] and still the same error

**Comment by elunenule** [comment]
Would this be related to py4J?

**Comment by elunenule** [comment]
I have tested with pyspark 2.2.2 and 2.2.3 with py4J 10.4.7 and i still have the same issue

**Comment by mn-mikke** [comment]
Hi @elunenule,
thanks for reporting the issue. This is a bug and will be fixed in the next fix release `3.32.0.3-1`. https://h2oai.atlassian.net/browse/SW-2501
